### DISCUSSION TITLE
x-bind:asp-action not working

### DISCUSSION THREAD

**Discussion by melakhvision** [alpinejs/alpine#3871] [7. Bugs (archived)]
Hello
I have added alpine js in my .NET MVC project. I manage to get it working expect the x-bind directive on asp-action.
I have test explicit and implicit directive but of them is working.
When I add the x-bind:asp-action and  x-bind:asp-controller in the dev console the value get populated but the it's not working
```hmtl
<template x-for="item in menuItems" :key="item.id">
    <li :key="item.id">
        <a x-bind:asp-controller="item.controller" x-bind:asp-action="item.action" class='text-uppercase' x-text="item.name"></a>
    </li>
</template>
```
This what I get in the dev console 
![SHOW TO TIMES](https://github.com/alpinejs/alpine/assets/45154979/51a2888d-18a3-4516-9a2a-b11357381908)

**Comment by ekwoka** [comment]
the .net asp attributes are preprocessed out of the html. They are not live on the page.

Per their docs:

The following markup submits the form to the Index action of HomeController when the input or button are selected:

```html
<form method="post">
    <button asp-controller="Home" asp-action="Index">Click Me</button>
    <input type="image" src="..." alt="Or Click Me" asp-controller="Home" 
                                asp-action="Index">
</form>
```

The previous markup generates following HTML:

```html
<form method="post">
    <button formaction="/Home">Click Me</button>
    <input type="image" src="..." alt="Or Click Me" formaction="/Home">
</form>
```

as you can see, the `asp` attributes are removed and other things are put in their place. No code runs on the client to handle this conversion.

**Comment by melakhvision** [comment]
How to solve or any workaround ?

**Reply by ekwoka** [reply]
You need to set the asp-acrion in the html.

As asp wants you to do.

Or otherwise create some data in the client that can create the same result.

**Reply by melakhvision** [reply]
@ekwoka The asp-action is set in the html as you see in the preview. I don't know what you mean

**Reply by ekwoka** [reply]
Where?

```html
<template x-for="item in menuItems" :key="item.id">
    <li :key="item.id">
        <a x-bind:asp-controller="item.controller" x-bind:asp-action="item.action" class='text-uppercase' x-text="item.name"></a>
    </li>
</template>
```

I see an `x-bind:asp-action`, I don't see an `asp-action`.

You should probably go and actually read the docs for this asp feature.

**Reply by melakhvision** [reply]
Yeah I set x-bind:asp-action. I cannot set  asp-action with an empty value. My gaol is to get the value from alpine binding

**Reply by ekwoka** [reply]
But Alpine runs in the client. `asp-action` is handled on the server

It's like trying to hardboil an egg before the chicken laid it.

**Reply by melakhvision** [reply]
asp-action handled on the server ? ASP .Net MVC is not server side for what I know maybe Blazor

**Reply by ekwoka** [reply]
https://learn.microsoft.com/en-us/aspnet/core/mvc/views/working-with-forms?view=aspnetcore-8.0

IDK what you think .net MVC is, but it's on the server, yes.
